app/upgrades/v7/upgrades.go

```diff
@@ -0,0 +1,183 @@
+)
+
+// Note: ensure these values are properly set before running upgrade
+var (
+	UpgradeName             = "v7"
+	IncentiveProgramAddress = "stride1tlxk4as9sgpqkh42cfaxqja0mdj6qculqshy0gg3glazmrnx3y8s8gsvqk"
+	StrideFoundationAddress = "stride1yz3mp7c2m739nftfrv5r3h6j64aqp95f3degpf"
+)
+
+// Helper function to log the migrated modules consensus versions
+func logModuleMigration(ctx sdk.Context, versionMap module.VersionMap, moduleName string) {
+	currentVersion := versionMap[moduleName]
+	ctx.Logger().Info(fmt.Sprintf("migrating module %s from version %d to version %d", moduleName, currentVersion-1, currentVersion))
+}
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v7
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	cdc codec.Codec,
+	epochsKeeper epochskeeper.Keeper,
+	stakeibcKeeper stakeibckeeper.Keeper,
+	icahostKeeper icahostkeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+	mintKeeper mintkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v7...")
+
+		// TODO:
+		//  	add min/max redemption rate
+		//      autopilot store key
+		//  	BaseAccount issue
+
+		// Add an hourly epoch which will be used by the rate limit store
+		AddHourEpoch(ctx, epochsKeeper)
+
+		// Increase stride inflation to 1078 STRD
+		IncreaseStrideInflation(ctx, mintKeeper)
+
+		// Add stride messages to the ICA host allow messages
+		AddICAHostAllowMessages(ctx, icahostKeeper)
+
+		// Add min/max redemption rate threshold for each host zone
+		AddMinMaxRedemptionRate(ctx, stakeibcKeeper)
+
+		// Change the juno unbonding frequency to 5
+		if err := ModifyJunoUnbondingFrequency(ctx, stakeibcKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to modify juno unbonding frequency")
+		}
+
+		// Incentive diversification
+		if err := IncentiveDiversification(ctx, bankKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to send ustrd tokens for prop #153 (incentive diversification)")
+		}
+
+		ctx.Logger().Info("Running module mogrations...")
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// Add a new hourly epoch that will be used by the rate limit module
+func AddHourEpoch(ctx sdk.Context, k epochskeeper.Keeper) {
+	ctx.Logger().Info("Adding hour epoch")
+
+	hourEpoch := epochstypes.EpochInfo{
+		Identifier:              epochstypes.HOUR_EPOCH,
+		StartTime:               time.Time{},
+		Duration:                time.Hour,
+		CurrentEpoch:            0,
+		CurrentEpochStartHeight: 0,
+		CurrentEpochStartTime:   time.Time{},
+		EpochCountingStarted:    false,
+	}
+
+	k.SetEpochInfo(ctx, hourEpoch)
+}
+
+// Increase stride inflation from 285 STRD to 1078 STRD
+func IncreaseStrideInflation(ctx sdk.Context, k mintkeeper.Keeper) {
+	ctx.Logger().Info("Increasing STRD inflation")
+
+	epochProvisions := sdk.NewDec(1_078_767_123_000_000)
```

Comment:
I think this should be 1_078_767_123, but please double check me on that!

Reply:
Good catch haha, you’re right!